lib/internal/queue.js

```diff
@@ -45,16 +47,14 @@ export default function queue(worker, concurrency, payload) {
     }
 
     function _next(tasks) {
-        return function(){
+        return rest(function(args){
             workers -= 1;
 
-            var removed = false;
-            var args = arguments;
             arrayEach(tasks, function (task) {
-                arrayEach(workersList, function (worker, index) {
-                    if (worker === task && !removed) {
+                arraySome(workersList, function (worker, index) {
```

Comment:
Note, this is a slight change in behaviour as more than 1 task can be removed per process. I think the fix should be only doing the rest of this functions processing in the outer arrayEach as we currently call tasks not in the workerList. I think we need to discuss this and identify some unit tests for this case